File: algorithm/구현/프로그래머스_크레인인형뽑기게임/프로그래머스_크레인인형뽑기게임.java
import java.util.*;

class Solution {
	public int solution(int[][] board, int[] moves) {
		int answer = 0;
		Stack<Integer> st = new Stack<>();
		for(int i=0; i<moves.length; i++){
			int loc = moves[i]-1;
			for(int j=0; j<board.length; j++){
				if(board[j][loc]!=0){
					if(!st.isEmpty() && st.peek()==board[j][loc]){
						st.pop();
						answer+=2;
					}
					else
						st.push(board[j][loc]);
					board[j][loc] = 0;
					break;
				}

			}
		}
		return answer;
	}
}

File: algorithm/구현/프로그래머스_크레인인형뽑기게임/프로그래머스_크레인인형뽑기게임.js
function solution(board, moves) {
    var answer = 0;
    
    var stack = []
    
    stack.push(1)
    stack.push(2)
    console.log(stack.pop())

    for(let j=0; j<moves.length; j++){
        for(let i=0; i<board.length; i++){
            if(board[i][moves[j]-1]!=0){
                if(stack[stack.length-1]==board[i][moves[j]-1]){
                    stack.pop()
                    answer+=2
                }else
                    stack.push(board[i][moves[j]-1])
                board[i][moves[j]-1]=0
                break;
            }
        }                
    }
    
    return answer;
}